### PR TITLE
add support for trailing commas in arrays

### PR DESCRIPTION
**Pull request by bmakuh**
Partially addresses https://github.com/grain-lang/grain/issues/182

### CODE REVIEW

File: compiler/src/parsing/parser.dyp [hunk 1]
```diff
@@ -540,9 +540,13 @@ list_expr :
   | lbrack rbrack { Exp.list ~loc:(symbol_rloc dyp) [] None }
   | lbrack expr [comma expr {$2}]* list_expr_ending rbrack { Exp.list ~loc:(symbol_rloc dyp) ($2::$3) $4 }
 
+array_expr_ending :
+  | comma? { None }
+  | comma ELLIPSIS expr { Some $3 }
+
```
Comment (ospencer) [inline compiler/src/parsing/parser.dyp:546]: We don't have array spreads, so might as well inline and save on a nonterminal:
```suggestion
```

File: compiler/src/parsing/parser.dyp [hunk 2]
```diff
@@ -540,9 +540,13 @@ list_expr :
...
 array_expr :
   | lbrack rcaret rbrack { Exp.array ~loc:(symbol_rloc dyp) [] }
-  | lbrack rcaret expr [comma expr {$2}]* rbrack { Exp.array ~loc:(symbol_rloc dyp) ($3::$4) }
+  | lbrack rcaret expr [comma expr {$2}]* array_expr_ending rbrack { Exp.array ~loc:(symbol_rloc dyp) ($3::$4) }
```
Comment (ospencer) [inline compiler/src/parsing/parser.dyp:549]:
```suggestion
  | lbrack rcaret expr [comma expr {$2}]* comma? rbrack { Exp.array ~loc:(symbol_rloc dyp) ($3::$4) }
```

**Review by ospencer — CHANGES_REQUESTED** [review]
Sweet! I left a thought about the `array_expr_ending` rule; let me know what you think.

Would you mind adding a test to the more basic set here as well? https://github.com/grain-lang/grain/blob/a474e1ea314d8d382ce6d92076f5e4ae38a9d6b4/compiler/test/test_end_to_end.re#L411

Here's an example of how we do it for lists:
https://github.com/grain-lang/grain/blob/a474e1ea314d8d382ce6d92076f5e4ae38a9d6b4/compiler/test/test_end_to_end.re#L401

**Comment by bmakuh** [comment]
Thanks for the review @ospencer! I added some e2e tests and removed the spread token.

**Review by ospencer — APPROVED** [review]
🙌 

Really happy to see someone take this on 🙂

**Pull request merged**